elastic/apm-server
# agentcfg: implement Elasticsearch fetcher

## Motivation/summary



To remove dependency on fleet, poll ES directly for agent central config. Serve the fetch requests locally from a cache instead of querying on demand

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes



1. Configure agent config in Kibana
2. Check APM server logs (debug logs) that refresh cache is running ok, and/or in stats `apm-server.agentcfg.elasticsearch`, confirm cache size `cache.entries.count` and cache refresh success `cache.refresh.successes`
3. Use an agent to fetch agent config and confirm that agent config is fetched successfully

## Related issues



Fixes #9642
